android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java

```diff
@@ -0,0 +1,65 @@
+package com.reactnativecommunity.webview;
+
+import android.widget.FrameLayout;
+
+import androidx.annotation.Nullable;
+
+import com.facebook.common.logging.FLog;
+import com.facebook.react.uimanager.ThemedReactContext;
+
+public class RNCWebView extends FrameLayout {
+  private static final String TAG = "RNCWebView";
+
+  private RNCWebViewManager.InternalWebView internalWebView;
+
+  public RNCWebView(ThemedReactContext reactContext) {
+    super(reactContext);
+  }
+
+  public interface Action {
+    void apply(RNCWebViewManager.InternalWebView webView);
+  }
+
+  public void attachWebView(RNCWebViewManager.InternalWebView webView) {
+    this.internalWebView = webView;
+
+    // Only re-attach the WebView if parent is null
+    if (webView.getParent() != null) {
+      throw new IllegalArgumentException("WebView with key: " + webView.webViewKey + " parent is non null. Cannot re-attach webview.");
+    }
+
+    // Fixes broken full-screen modals/galleries due to body height being 0.
+    addView(webView, new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
+  }
+
+  public RNCWebViewManager.InternalWebView detachWebView() {
+    if (internalWebView == null) {
+      throw new IllegalStateException("Webview is null");
+    }
+
+    removeWebViewFromParent();
+    RNCWebViewManager.InternalWebView webView = internalWebView;
+    this.internalWebView = null;
+    return webView;
+  }
+
+  public void removeWebViewFromParent() {
+    if (internalWebView != null) {
+      removeView(internalWebView);
+    }
+  }
+
+  @Nullable
+  public RNCWebViewManager.InternalWebView getWebView() {
+    return internalWebView;
+  }
+
+  public void ifPresent(Action action) {
```

Comment:
@donaldchen this is my proposed approach of handling nullability

Reply:
Nice! This looks clean to me

Reply:
nit on naming: would something like `ifHasInternalWebView` instead of `ifPresent` be clearer? (I'm also okay with `ifPresent`)

Reply:
I think this is somewhat a convention of Optional types in Java. https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#ifPresent-java.util.function.Consumer-

At the same time, it's readable for all `ifHasInternalWebView` whether you know the above convention or not. I'll update to what you have suggested 👍

Reply:
Ah I didn't know about that Java convention. Thanks for the link.

> At the same time, it's readable for all ifHasInternalWebView whether you know the above convention or not. I'll update to what you have suggested 👍

👍

Reply:
done